packages/graphql-server-express/src/expressApollo.ts

```diff
@@ -42,33 +42,48 @@ export function graphqlExpress(options: GraphQLOptions | ExpressGraphQLOptionsFu
     }
 
     const formatErrorFn = optionsObject.formatError || graphql.formatError;
+    let requestPayload;
+
+    switch ( req.method ) {
+      case 'POST':
+        if ( !req.body ) {
+          res.statusCode = 500;
+          res.write('POST body missing. Did you forget "app.use(bodyParser.json())"?');
+          res.end();
+          return;
+        }
 
-    if (req.method !== 'POST') {
-      res.setHeader('Allow', 'POST');
-      res.statusCode = 405;
-      res.write('Apollo Server supports only POST requests.');
-      res.end();
-      return;
-    }
+        requestPayload = req.body;
+        break;
+      case 'GET':
+        if ( !req.query ) {
+          res.statusCode = 500;
```

Comment:
ohhh
i think you are abit confuse :)

if you will take a deeper look, req.query for GET is equal to req.body of POST.
which means, in req.query you will get all the query params as key value.
(?key=value&seckey=value) will become:
```
{
  "key": "value",
  "seckey": "value",
}
```

if i was checking req.query.query, you were correct on your assumption.
is it more clear now? do you still think it should be 400 and not 500?